### PR TITLE
Handle out of storage error from driver and make the container loadable

### PR DESCRIPTION
**Pull request by jatgarg**
## Description

Handle out of storage error from driver and make the container loadable. Currently there is no handling and when the join session throws this error, the container will get closed. With this we use `NoDeltaStream` object as connection and load the container in read mode, so that it loads properly. We also notify the `"readonly"` through the event on delta manager so that apps can listen to this and show any UX etc. The app can listen to 

```
container.deltaManager.on("readonly", (readonly?: boolean, reason?: string) => {
           // reason will be equal to DriverErrorType.outOfStorage in this case
          // App logic
 }

```
Also deprecated the `OdspErrorType.outOfStorage` in favour of `DriverErrorType.outOfStorage` as this could be a general error and we can therefore move it out of ODSP.

### CODE REVIEW

**Review by AndreiZe — APPROVED** [review]
LGTM

File: packages/loader/container-loader/src/connectionManager.ts [hunk 1]
```diff
@@ -486,10 +497,11 @@ export class ConnectionManager implements IConnectionManager {
 	private set_readonlyPermissions(
 		newReadonlyValue: boolean,
 		oldReadonlyValue: boolean | undefined,
+		reason?: string,
```
Comment (anthony-murphy) [inline packages/loader/container-loader/src/connectionManager.ts:500]: consider using IConnectionStateChangeReason here, so errors can be tracked as well

File: packages/loader/container-loader/src/connectionManager.ts [hunk 2]
```diff
@@ -804,7 +825,11 @@ export class ConnectionManager implements IConnectionManager {
 			0x0e8 /* "readonly perf with write connection" */,
 		);
 
-		this.set_readonlyPermissions(readonly, oldReadonlyValue);
+		this.set_readonlyPermissions(
+			readonly,
+			oldReadonlyValue,
+			isNoDeltaStreamConnection(connection) ? connection.error?.errorType : undefined,
```
Comment (anthony-murphy) [inline packages/loader/container-loader/src/connectionManager.ts:831]: i would also use IConnectionStateChangeReason in NoDeltaStreamConnection, se we track and propagate both error and reason

**Comment by jatgarg** [comment]
I would also open this PR in next branch and close this one.

**Pull request closed without merging**